core/types/block_ext_test.go

```diff
@@ -106,3 +111,410 @@ func assertDifferentPointers[T any](t *testing.T, a *T, b any) {
+
+	uncle := &Header{ParentHash: common.Hash{6}}
+	uncleExtra := &HeaderExtra{ExtDataHash: common.Hash{7}}
+	SetHeaderExtra(uncle, uncleExtra)
+
+	body := &Body{
+		Transactions: []*Transaction{testTx},
+		Uncles:       []*Header{uncle},
+		Withdrawals:  []*ethtypes.Withdrawal{{Index: 7}}, // ignored
+	}
+	extra := &BlockBodyExtra{
+		Version: 1,
+		ExtData: ptrTo([]byte{2}),
+	}
+	extras.Body.Set(body, extra)
+
+	b, err := rlp.EncodeToBytes(body)
+	require.NoError(t, err)
+
+	gotBody := new(Body)
+	err = rlp.DecodeBytes(b, gotBody)
+	require.NoError(t, err)
+
+	// Check transactions in the following because unexported fields
+	// size and time are set by the RLP decoder.
+	require.Len(t, gotBody.Transactions, 1)
+	gotTx := gotBody.Transactions[0]
+	gotTxJSON, err := gotTx.MarshalJSON()
+	require.NoError(t, err)
+	wantTxJSON, err := testTx.MarshalJSON()
+	require.NoError(t, err)
+	assert.Equal(t, string(wantTxJSON), string(gotTxJSON))
+	gotBody.Transactions = nil
+
+	wantUncle := &Header{
+		ParentHash: common.Hash{6},
+		Difficulty: new(big.Int),
+		Number:     new(big.Int),
+		Extra:      []byte{},
+	}
+	wantUncleExtra := &HeaderExtra{ExtDataHash: common.Hash{7}}
+	SetHeaderExtra(wantUncle, wantUncleExtra)
+
+	wantBody := &Body{
+		Transactions: nil, // checked above
+		Uncles:       []*Header{wantUncle},
+	}
+	wantExtra := &BlockBodyExtra{
+		Version: 1,
+		ExtData: ptrTo([]byte{2}),
+	}
+	extras.Body.Set(wantBody, wantExtra)
+
+	assert.Equal(t, wantBody, gotBody)
+	gotExtra := extras.Body.Get(gotBody)
+	assert.Equal(t, wantExtra, gotExtra)
+}
+
+func TestBlockExtraRLP(t *testing.T) {
+	t.Parallel()
+
+	header := &Header{ParentHash: common.Hash{1}}
+	headerExtras := &HeaderExtra{ExtDataHash: common.Hash{2}}
+	SetHeaderExtra(header, headerExtras)
+
+	testTx := NewTransaction(3, common.Address{4}, big.NewInt(2), 3, big.NewInt(4), []byte{5})
+	txs := []*Transaction{testTx}
+
+	uncle := &Header{ParentHash: common.Hash{6}}
+	uncleExtra := &HeaderExtra{ExtDataHash: common.Hash{7}}
+	SetHeaderExtra(uncle, uncleExtra)
+	uncles := []*Header{uncle}
+
+	receipts := []*Receipt{{PostState: []byte{8}}}
+
+	block := NewBlock(header, txs, uncles, receipts, stubHasher{})
+
+	withdrawals := []*ethtypes.Withdrawal{{Index: 9}} // ignored
+	block = block.WithWithdrawals(withdrawals)
+
+	extra := &BlockBodyExtra{
+		Version: 10,
+		ExtData: ptrTo([]byte{11}),
+	}
+	extras.Block.Set(block, extra)
+
+	b, err := rlp.EncodeToBytes(block)
+	require.NoError(t, err)
+
+	gotBlock := new(Block)
+	err = rlp.DecodeBytes(b, gotBlock)
+	require.NoError(t, err)
+
+	// Check transactions in the following because unexported fields
+	// size and time are set by the RLP decoder.
+	gotTxs := gotBlock.Transactions()
+	require.Len(t, gotTxs, 1)
+	gotTx := gotTxs[0]
+	gotTxBin, err := gotTx.MarshalBinary()
+	require.NoError(t, err)
+	wantTxBin, err := testTx.MarshalBinary()
+	require.NoError(t, err)
+	assert.Equal(t, wantTxBin, gotTxBin)
```

Comment:
See my [comment on the equivalent code above](https://github.com/ava-labs/coreth/pull/760/files#r1970522666).

Reply:
Tests refactored and this is no longer needed, resolving this!